controllers/main.js

```diff
@@ -29,7 +29,7 @@ const RenderHome = async (_req, res) => {
     let today = (new Date()).getDay();
     const tt = await FetchTimetable('class-4');
     let data=[];
-    if (tt){
+    if (tt && tt.schedule && tt.schedule.hasOwnProperty(today)){
```

Comment:
I am not sure of the use of hasOwnProperty() in this case.

Here, schedule is an array and today is the index which we are trying to access. Does hasOwnProperty() work for arrays as well?

Reply:
My bad. Actually in the scenario where mongodb connection fails, you will get the following error,
```
data = tt.schedule[today].data; 
                ^
TypeError: Cannot read property '2' of undefined
```
Therefore the correction should be,
```
if (tt){
        data = tt.schedule.length>today?tt.schedule[today].data:[];
}
```